compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1118,7 +1118,7 @@ class Definitions {
         case ErasedFunctionOf(mt) =>
           Some(mt.paramInfos, mt.resType, mt.isContextualMethod)
         case _ =>
-          val tsym = ft.dealias.typeSymbol
+          val tsym = ft.typeSymbol
```

Comment:
What was the type of the ` ft` that caused the issue and what was the dealiased type?

Reply:
Maybe we should figure out what this has in common with https://github.com/lampepfl/dotty/issues/18276. I'm currently suspecting a change in Typer in https://github.com/lampepfl/dotty/pull/16507/files#diff-8c9ece1772bd78160fc1c31e988664586c9df566a1d22ff99ef99dd6d5627a90 could have broken this.

Reply:
I think they are mostly unrelated unfortunately, [this](https://github.com/lampepfl/dotty/pull/16507/files#diff-91bb9b80218c5b1c9fb90d562a52c1aa7a271d795ffffa86c9cd8d1cc133e5ffR1107) was the line that broke #17245, but It does not seem to have any effect on #18276 in my tests.

Reply:
Indeed, they are unrelated. It fixed the other one (waiting for CI confirmation) in #18288.

Reply:
Now I am wondering if you need #18288 to fix the new issue you are having.

Reply:
Just tested now and it does not seem to affect it, but keeping the dealiasing in FunctionOf unapply and instead changing the `resolveOverloaded1` method [here](https://github.com/lampepfl/dotty/blob/45e08cb968fe75e2ee47aca5375e6f2acd0aadde/compiler/src/dotty/tools/dotc/typer/Applications.scala#L2062C8-L2062C8) seems to work for all of the now-failing tests, even if changing only one line very naively, so we can skip that case which previously would not be accessed, like this: 
```diff
- case defn.FunctionOf(args, resultType, _) =>
+ case defn.FunctionOf(args, resultType, _) if pt.dealias.typeSymbol == pt.typeSymbol =>
```
But I want to read up on this method some more today in case there is a cleaner solution, and then I will update this PR by tomorrow morning